[24.10] luci-base: Hardcode 'openwrt-24.10' as the branch name

hnyman opened the pull request:
Hardcode 'openwrt-24.10' release branch as the branch name to avoid `LuCI (HEAD detached at 2ac26e56) branch` being shown as the branch.

Feeds are since March 2025 cloned shallow if cloned from a specific commit, like done with a release tag. Change in OpenWrt: https://github.com/openwrt/openwrt/commit/32d0a57dc10811a24816e51f10e55963f40fe462

As the only cloned commit of a release tag is without history, the branch can't be identified.

So, hardcode the branch name to get the proper branch name `LuCI openwrt-24.10 branch` into version.uc.

Fixes #7737

Similar change should likely be done also for 23.05 (into where the same feeds logic has been backported).

Note that the change will only have impact from 24.10.2 onward, as 24.10.1 tag is naturally unimpacted by this.

The change is unnecessary for builds from feeds with full history, but locking the release branch name is harmless, as the branch name does not change.

cc @robimarko @subnut @4nth5 @jow- 

----
Before:
```
perus@ub2504:/OpenWrt/r7800-2410$ make package/luci-base/clean ; make -j 5 package/luci-base/compile ; cat staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/root-ipq806x/usr/share/ucode/luci/version.uc  ; (cd feeds/luci ; git status | head -n 3 ; git diff )
 make[1] package/luci-base/compile
...
 make[2] -C feeds/luci/libs/rpcd-mod-luci compile
 make[2] -C feeds/luci/modules/luci-base compile
export const revision = '25.103.51521~2ac26e5', branch = 'LuCI (HEAD detached at 2ac26e56) branch';
HEAD detached at 2ac26e56
nothing to commit, working tree clean
```

With this PR:
```
perus@ub2504:/OpenWrt/r7800-2410$ make package/luci-base/clean ; make -j 5 package/luci-base/compile ; cat staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/root-ipq806x/usr/share/ucode/luci/version.uc  ; (cd feeds/luci ; git status | head -n 3 ; git diff )
...
 make[2] -C feeds/luci/modules/luci-base compile
export const revision = '25.103.51521~2ac26e5', branch = 'LuCI openwrt-24.10 branch';
HEAD detached at 2ac26e56
Changes not staged for commit:
```

systemcrash commented:
Right. So this should only exist in the relevant branches (and never in master)?

robimarko commented:
Honestly, is it worth the time to even remember to this for each release to avoid a cosmetic only thing?

systemcrash commented:
I don't think this is just an optics thing - it's about perception. It's a subtle change that some users might perceive as a bug because something looks different to normal, especially with snapshots in mind.

systemcrash commented:
Granted, that I also dislike extra burden to remember to do with each release 😮‍💨

systemcrash commented:
So... git is a versatile tool. And if we're in some kind of detached state, we can still figure out what branch contains that commit. hashes change when they're cherry picked (altho a CP is not guaranteed, but is usually the case after release). Doesn't it make sense to do something like:

```
git branch --contains HEAD
# or
git rev-parse HEAD
git branch --contains <commit-hash>
```

?

hnyman commented:
> Doesn't it make sense to do something like:

Note that the whole LuCI repo in case of a tag is a shallow clone. To my understanding no history and branches are available.
(It is not only about detached checkout of a commit from a full repo, but that the repo has been created/cloned to be just a single commit to avoid bandwidth usage for transferring history.

robimarko commented:
Wait a second, but shallow clone still provides us with the commit hash as well as the branch info.
For example, this is what I have locally on my test 24.10:
```
[robimarko@fedora luci]$ git log
commit a4d988d76be2b43bca66d1b937319433249b4c11 (grafted, HEAD -> openwrt-24.10, origin/openwrt-24.10)
Author: Eric Fahlgren <ericfahlgren@gmail.com>
Date:   Sun Apr 20 16:12:00 2025 -0700

    luci-app-attendedsysupgrade: change error reporting url
    
    Direct error reports to a LuCI-ASU-app-specific thread on the forum, where
    they'll get more attention and reduce clutter on the ASU server's github issues.
    
    Signed-off-by: Eric Fahlgren <ericfahlgren@gmail.com>
    (cherry picked from commit 9fd692ab4402260bbbe5512cfeff6427c18d39b4)
[robimarko@fedora luci]$ git branch --contains HEAD
* openwrt-24.10
```

hnyman commented:
Well, this is what happens if you really start by cloning a fresh OpenWrt and checking out the v24.10.1 tag and only then updating feeds...

```
perus@ub2504:/tmp$ git clone https://github.com/openwrt/openwrt.git
Cloning into 'openwrt'...
remote: Enumerating objects: 737021, done.
remote: Counting objects: 100% (23/23), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 737021 (delta 7), reused 10 (delta 4), pack-reused 736998 (from 3)
Receiving objects: 100% (737021/737021), 257.42 MiB | 28.54 MiB/s, done.
Resolving deltas: 100% (493318/493318), done.
perus@ub2504:/tmp$ cd openwrt/
perus@ub2504:/tmp/openwrt$ git fetch --tags
perus@ub2504:/tmp/openwrt$ git checkout v24.10.1
Note: switching to 'v24.10.1'.
...
HEAD is now at d183d7bb78 OpenWrt v24.10.1: adjust config defaults

perus@ub2504:/tmp/openwrt$ ./scripts/feeds update
Updating feed 'packages' from 'https://git.openwrt.org/feed/packages.git^d8cd30f4e281d6853b3de134c4f147a807583e43' ...
Cloning into './feeds/packages'...
remote: Enumerating objects: 7329, done.
...
Resolving deltas: 100% (378/378), completed with 373 local objects.
From https://git.openwrt.org/feed/packages
 * branch            d8cd30f4e281d6853b3de134c4f147a807583e43 -> FETCH_HEAD
HEAD is now at d8cd30f lua-eco: update to 3.8.0
/tmp/openwrt
Updating feed 'luci' from 'https://git.openwrt.org/project/luci.git^2ac26e56cc55102cb10e7b0867c2b78e0f6d5fd8' ...
Cloning into './feeds/luci'...
remote: Enumerating objects: 10890, done.
remote: Counting objects: 100% (10890/10890), done.
remote: Compressing objects: 100% (5076/5076), done.
remote: Total 10890 (delta 3633), reused 7159 (delta 974), pack-reused 0
Receiving objects: 100% (10890/10890), 10.12 MiB | 7.49 MiB/s, done.
Resolving deltas: 100% (3633/3633), done.
remote: Enumerating objects: 8272, done.
remote: Counting objects: 100% (8272/8272), done.
remote: Compressing objects: 100% (1516/1516), done.
remote: Total 4681 (delta 2253), reused 3262 (delta 946), pack-reused 0
Receiving objects: 100% (4681/4681), 2.02 MiB | 4.90 MiB/s, done.
Resolving deltas: 100% (2253/2253), completed with 1423 local objects.
From https://git.openwrt.org/project/luci
 * branch              2ac26e56cc55102cb10e7b0867c2b78e0f6d5fd8 -> FETCH_HEAD
HEAD is now at 2ac26e56 luci-proto-wireguard: fixed incorrect peer name detection
/tmp/openwrt
Updating feed 'routing' from 'https://git.openwrt.org/feed/routing.git^c9b636698881059a3c981032770968f5a98ff201' ...
Cloning into './feeds/routing'...
remote: Enumerating objects: 288, done.
...
Unpacking objects: 100% (31/31), 14.92 KiB | 954.00 KiB/s, done.
From https://git.openwrt.org/feed/routing
 * branch            c9b636698881059a3c981032770968f5a98ff201 -> FETCH_HEAD
HEAD is now at c9b6366 Merge pull request #1104 from ecsv/batadv-for-24.10
/tmp/openwrt
Updating feed 'telephony' from 'https://git.openwrt.org/feed/telephony.git^fd605af7143165a2490681ec1752f259873b9147' ...
Cloning into './feeds/telephony'...
remote: Enumerating objects: 263, done.
...
Unpacking objects: 100% (68/68), 57.35 KiB | 3.82 MiB/s, done.
From https://git.openwrt.org/feed/telephony
 * branch            fd605af7143165a2490681ec1752f259873b9147 -> FETCH_HEAD
HEAD is now at fd605af CI: multi-arch-test-build: move to shared workflow
/tmp/openwrt
Create index file './feeds/packages.index' 
Checking 'true'... ok.
...
Checking 'libintl.h'... ok.
Checking 'ldconfig-stub'... ok.
...
Collecting package info: done
Collecting target info: done

perus@ub2504:/tmp/openwrt$ cd feeds/luci

perus@ub2504:/tmp/openwrt/feeds/luci$ git log
commit 2ac26e56cc55102cb10e7b0867c2b78e0f6d5fd8 (grafted, HEAD)
Author: this-username-has-been-taken <this_username_has_been_taken2@proton.me>
Date:   Sun Apr 13 16:38:46 2025 +0300

    luci-proto-wireguard: fixed incorrect peer name detection
    
    Added the legacy peer name detection algorithm. If the new algorithm
    fails to find a peer's name (e.g. when the IP address has been changed)
    then the legacy value will be used.
    
    Signed-off-by: Tom Haley <this_username_has_been_taken2@proton.me>
    (cherry picked from commit 8bff54dc47057e36c0ceb263efe4c82580003924)

perus@ub2504:/tmp/openwrt/feeds/luci$ git branch --contains HEAD
* (HEAD detached at 2ac26e56)

```

No branch...

robimarko commented:
Yeah, when using tags then there is no info.

I am fine with whatever you guys decide.

systemcrash commented:
Aaah, ofc. git elides all of the branch info from the repo. And it's just a mono-commit. :/

OK.

subnut commented:
I was tagged/mentioned in this thread.

I am terribly sorry but I currently won't be able to comment on anything, as I have been in an ICU (Intensive Care Unit) for the last three days.

I hope everybody is having a good day.
S.
